compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/function-expression-prototype-call.expect.md

```diff
@@ -23,11 +23,11 @@ import { c as _c } from "react/compiler-runtime";
 function Component(props) {
   const $ = _c(4);
   let t0;
-  if ($[0] !== props.name) {
+  if ($[0] !== props) {
```

Comment:
we could teach the compiler about Function.prototype.call/apply but this is likely super rare, this makes sense

Reply:
Does JSX allow you to pass in null for props? I don't know what the syntax for that would be. I think you'd need to invoke createElement/jsx directly to get a nullable props argument.

Do we actually need to be more conservative on props?

Reply:
good catch, i thought we already tracked non-null objects. maybe the changes here are overriding that?